srtcore/buffer.cpp

```diff
@@ -63,6 +63,41 @@ using namespace std;
 using namespace srt_logging;
 using namespace srt::sync;
 
+bool AvgBufSize::canUpdate(const time_point& now) const
+{
+    const int      SRT_MAVG_BASE_PERIOD = 1000000; // us
+    const int      SRT_us2ms            = 1000;
+    const uint64_t elapsed_ms           = count_milliseconds(now - m_tsLastSamplingTime); // ms since last sampling
+    return (elapsed_ms >= (SRT_MAVG_BASE_PERIOD / SRT_MAVG_SAMPLING_RATE) / SRT_us2ms);
+}
+
+void AvgBufSize::update(const steady_clock::time_point& now, int pkts, int bytes, int timespan_ms)
+{
+    const uint64_t elapsed_ms = count_milliseconds(now - m_tsLastSamplingTime); // ms since last sampling
+
+    m_tsLastSamplingTime = now;
+
+    if (elapsed_ms > 1000)
```

Comment:
Similar here: `now - m_tsLastSamplingTime > milliseconds_from(1000)` ?

Reply:
We already have `elapsed_ms`. It is later used for the interval-based moving average.
I would not like to perform the same operation twice.
It is probably worth to rather replace 1000 with a constant.